pkg/api/v1/types.go

```diff
@@ -116,6 +119,7 @@ type VMList struct {
 
 // VMSpec is a description of a VM
 type VMSpec struct {
+	// Domain is the actual libvirt domain in JSON format. When creating a VM, it is converted to XML.
```

Comment:
I would not mention a concrete format here (e.g. YAML is also possible, and sooner or later protobuf). Instead of pointing out the difference here, I would prefer if you would add the comment, to avoid confiusions with virt-handlers api.DomainSpec on the struct, rather than on the point where it is used.